### DISCUSSION TITLE
Specific time steps' output

### DISCUSSION THREAD

**Discussion by hokonkim** [idaholab/moose#21043] [Q&A General] [answered]
Hello, I have a question about output.

I set the syntax for output like below.
    
    [Outputs]
      exodus = true
      [csv]
        type = CSV
        file_base = outputs/cathode_test
        append_date = true
        append_date_format = '%Y-%m-%d-%R-%S'
      []
      [exodus1]
        type = Exodus
        file_base = outputs/cathode_test1
        append_date = true
        append_date_format = '%Y-%m-%d-%R-%S'
        start_time = 0
        interval = 100
        execute_on = 'timestep_end'
      []
    []

For example, this simulation give me results from time = -1 to time = 491.

What I want to do is "output the results at specific time (time = 0, 100, 200, 300, 400, 491(last time step)).

Plus, currently always all output result file was generated as well. This is too big, so I don't want this to be generated.

How can I get this specific output file without generating the file including all results.

Thank you.

Best regards,
Hokon Kim

**Comment by GiudGiud** [comment, marked as answer]
Hello

You should use the interval parameter of the output. And set it to 100

Then you also want to execute on final to catch this 491 timestep
see these docs, there's an example in the `Multiple Output Blocks` paragraph or in `Common parameters`
https://mooseframework.inl.gov/syntax/Outputs/index.html

Guillaume

**Reply by GiudGiud** [reply]
actually it seems you are already using this parameter. 
What are you seeing? THe interval is not respected?

**Reply by hokonkim** [reply]
time = 99, 199, 299, 399

Only four time steps are stored in the result output(*.e).

I thought I could get the results when time = 0, 100, 200, 300, 400, ...

**Reply by GiudGiud** [reply]
If you want to offset it by one you should move the start_time of the output. 

Yeah adding on final might be tricky with the current capability. I hope execute_on = 'timestep_end final' will do it but I suspect the interval is obeyed first and foremost

**Reply by hokonkim** [reply]
I still don't get it about Output options.

Options doesn't work as I wanted.

    [Outputs]
      [csv]
        type = CSV
        file_base = outputs/cathode_test
        append_date = true
        append_date_format = '%Y-%m-%d-%R-%S'
        execute_on = 'initial timestep_end'
      []
      [exo]
        type = Exodus
        file_base = outputs/cathode_test1
        append_date = true
        append_date_format = '%Y-%m-%d-%R-%S'
        start_time = 0
        interval = 5
      []
    []

I expected that time = 0, 5, 10, ..., but the result really I got was "time = 4, 9, 14".

I think the interval option did not work properly as I wanted.

I thought "start_time + interval", but actually it didn't work like this.
(e.g, start_time = 1, interval = 3, then 1, 4, 7, 10, ..., but it is not...)

Interval just printed out the result at "N"th time step whatever start_time is...

**Reply by cpgr** [reply]
Yeah, I think the interval param just outputs every n steps, not every n time units. Having output every n time units would be a welcome addition. In the past, I've had to use `sync_times` and provide a list of times, which can be tedious!

**Reply by hokonkim** [reply]
I think `sync_times` is the proper way for my results file, currently.

Thanks for your reply!
